File: terminal/src/ansi.ts
export const isANSISupported = true;
// export const isANSISupported = false; // ANSI support is not guaranteed in Windows

const RESET = "\x1b[0m";
const HIDE_CURSOR = "\x1b[?25l";
const SHOW_CURSOR = "\x1b[?25h";
const CURSOR_HOME = "\x1b[H";
const CLEAR = "\x1b[2J";

export function clearAndDrawBuffer(buffer: string) {
    if (!isANSISupported)
        return;

    process.stdout.write(CLEAR + HIDE_CURSOR + CURSOR_HOME + buffer + RESET);
}

export function cleanup() {
    if (!isANSISupported)
        return;

    process.stdout.write(CLEAR + CURSOR_HOME + SHOW_CURSOR);
}

File: terminal-go/ansi/ansi.go
package ansi

import "fmt"

// TODO : test for CMD and PowerShell support, eg. forcing ANSI support in Windows
const isAnsiSupported = true

const RESET = "\x1b[0m"
const HIDE_CURSOR = "\x1b[?25l"
const SHOW_CURSOR = "\x1b[?25h"
const CURSOR_HOME = "\x1b[H"
const CLEAR = "\x1b[2J"

func IsAnsiSupported() bool {
	return isAnsiSupported
}

func ClearAndDrawBuffer(buffer string) {
	if !isAnsiSupported {
		return
	}

	const hideCursor = true
	// const hideCursor = false

	if hideCursor {
		fmt.Print(CLEAR + HIDE_CURSOR + CURSOR_HOME + buffer + RESET)
	} else {
		fmt.Print(CLEAR + CURSOR_HOME + buffer + RESET)
	}
}

func Cleanup() {
	if !isAnsiSupported {
		return
	}

	// fmt.Println("cleanup")

	const clear = true
	// const clear = false

	if clear {
		fmt.Print(CLEAR + CURSOR_HOME + SHOW_CURSOR)
	} else {
		fmt.Print(SHOW_CURSOR)
	}
}
